File: VB/TreeListDrawImage/DrawHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.ViewInfo
Imports DevExpress.XtraTreeList.Nodes
Imports System.Windows.Forms
Imports System.Drawing

Namespace TreeListDrawImage
    Public Class DrawHelper
        Private TreeList As TreeList

        Public Sub RegisterTreeList(ByVal treeList As TreeList)
            Me.TreeList = treeList
            SubscribeEvents()
        End Sub

        Public Sub RegisterTreeList(ByVal treeList As TreeList, ByVal image As Image)
            RegisterTreeList(treeList)
            Me.Image = image
        End Sub

        Public Property Image() As Image

        Public Event ImageClick As EventHandler

        Private Sub SubscribeEvents()
            AddHandler TreeList.CustomDrawNodeIndicator, AddressOf treeList1_CustomDrawNodeIndicator
            AddHandler TreeList.Click, AddressOf treeList1_Click
            AddHandler TreeList.MouseMove, AddressOf treeList1_MouseMove
        End Sub

        Private Sub treeList1_CustomDrawNodeIndicator(ByVal sender As Object, ByVal e As CustomDrawNodeIndicatorEventArgs)
            TreeList.ViewInfo.PaintAppearance.Row.FillRectangle(e.Cache, e.Bounds)
            If (TryCast(TreeList.ViewInfo.RowsInfo(e.Node), RowInfo)).Bounds.Contains(TreeList.PointToClient(Control.MousePosition)) Then
                e.Graphics.DrawImage(Image, e.Bounds.Location)
            End If
            e.Handled = True
        End Sub

        Private Sub treeList1_Click(ByVal sender As Object, ByVal e As EventArgs)
            If TreeList.CalcHitInfo(TreeList.PointToClient(Control.MousePosition)).HitInfoType = HitInfoType.RowIndicator Then
                RaiseEvent ImageClick(Me, EventArgs.Empty)
            End If
        End Sub

        Private node As TreeListNode
        Private Sub treeList1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim currentNode As TreeListNode = TreeList.CalcHitInfo(TreeList.PointToClient(Control.MousePosition)).Node
            If currentNode IsNot node Then
                node = currentNode
                TreeList.InvalidateNodes()
            End If
        End Sub
    End Class
End Namespace

File: CS/TreeListDrawImage/DrawHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.ViewInfo;
using DevExpress.XtraTreeList.Nodes;
using System.Windows.Forms;
using System.Drawing;

namespace TreeListDrawImage
{
    public class DrawHelper
    {
        TreeList TreeList;

        public void RegisterTreeList(TreeList treeList)
        {
            TreeList = treeList;
            SubscribeEvents();
        }

        public void RegisterTreeList(TreeList treeList, Image image)
        {
            RegisterTreeList(treeList);
            Image = image;
        }

        public Image Image { get; set; }

        public event EventHandler ImageClick;

        private void SubscribeEvents()
        {
            TreeList.CustomDrawNodeIndicator += treeList1_CustomDrawNodeIndicator;
            TreeList.Click += treeList1_Click;
            TreeList.MouseMove += treeList1_MouseMove;
        }

        private void treeList1_CustomDrawNodeIndicator(object sender, CustomDrawNodeIndicatorEventArgs e)
        {
            TreeList.ViewInfo.PaintAppearance.Row.FillRectangle(e.Cache, e.Bounds);
            if ((TreeList.ViewInfo.RowsInfo[e.Node] as RowInfo).Bounds.Contains(TreeList.PointToClient(Control.MousePosition)))
            {
                e.Graphics.DrawImage(Image, e.Bounds.Location);
            }
            e.Handled = true;
        }

        private void treeList1_Click(object sender, EventArgs e)
        {
            if (TreeList.CalcHitInfo(TreeList.PointToClient(Control.MousePosition)).HitInfoType == HitInfoType.RowIndicator)
            {
                if (ImageClick != null)
                    ImageClick(this, EventArgs.Empty);
            }
        }

        TreeListNode node;
        private void treeList1_MouseMove(object sender, MouseEventArgs e)
        {
            TreeListNode currentNode = TreeList.CalcHitInfo(TreeList.PointToClient(Control.MousePosition)).Node;
            if (currentNode != node)
            {
                node = currentNode;
                TreeList.InvalidateNodes();
            }
        }
    }
}
